test/integration/integration.test.ts

```diff
@@ -1135,3 +1135,320 @@ test('calls eslintrc function with repository and its location', async () => {
         "
     `);
 });
+
+describe('some tests with typescript configs', () => {
```

Comment:
Looks good, but let's replace these with a single test which validates we can actually pass Typescript into the configuration file, and see it executing:

```ts
test('loads typescript configuration file', async () => {
    const { output } = await runProductionBuild(
        {
            onComplete: `function onComplete() {
                console.log('[TEST-ON-COMPLETE-START]');

                type SomeType = 'a' | 'b' | 'c';
                const value: SomeType = 'b';
                console.log('Value of typed const', value);

                console.log('config file:');
                const index = process.argv.findIndex(a => a === '--config');
                const config = process.argv[index + 1];
                console.log(require('fs').readFileSync(config, 'utf8'));

                console.log('[TEST-ON-COMPLETE-END]');
            }` as unknown as Config['onComplete'],
        },
        undefined,
        'ts'
    );

    const [onCompleteCall] = output
        .join('\n')
        .match(/\[TEST-ON-COMPLETE-START\]([\s|\S]*)\[TEST-ON-COMPLETE-END\]/)!;

    expect(onCompleteCall).toMatchInlineSnapshot();
});
```

If typescript compiling is removed from code base we will see following error:
```ts
    expect(output).toMatchInlineSnapshot(`
        Array [
          "<removed>/test/integration/integration.config-0.js",
          "                type SomeType = 'a' | 'b' | 'c';",
          "                     ^^^^^^^^",
          "SyntaxError: Unexpected identifier",
          "    at Object.compileFunction (node:vm:352:18)",
          "    at wrapSafe (node:internal/modules/cjs/loader:1031:15)",
          "    at Module._compile (node:internal/modules/cjs/loader:1065:27)",
          "    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1153:10)",
          "    at Module.load (node:internal/modules/cjs/loader:981:32)",
          "    at Function.Module._load (node:internal/modules/cjs/loader:822:12)",
          "    at Module.require (node:internal/modules/cjs/loader:1005:19)",
          "    at require (node:internal/modules/cjs/helpers:94:18)",
          "    at loadConfig (<removed>/dist/config/load.js)",
          "    at Object.<anonymous> (<removed>/dist/config/config.js)",
        ]
    `);
```